src/usr/share/batwatch/batwatch.py

```diff
@@ -53,124 +46,150 @@ def __eq__(self, other):
-                self.logger.warn('A change in the battery status was detected. \nNEW STATUS: {} \nOLD STATUS: {}'
-                                 .format(current_status, new_status))
-                self._send_status_email(current_status, new_status)
-                current_status = new_status
+            current_status = self._get_composite_status()
+            if prior_status != current_status:
+                if prior_status < current_status:
+                    self.logger.warning('Battery state changed from %s to %s.',
+                                        prior_status, current_status)
+                else:
+                    self.logger.info('Battery state changed from %s to %s.', prior_status,
+                                     current_status)
+
+                self._send_email(self._build_email_body(prior_status, current_status))
+                prior_status = current_status
             else:
                 self.logger.trace('No changes in battery status.')
-            time.sleep(int(self.config['delay']))
 
-    def _get_composite_status(self):
-        """
-        Get status information about batteries connected to the device Batwatch is running on,
-         including the charge status, such that:
-        * If any battery is discharging, charge status is Discharging.
-          (Assume any status other than Charging or Fully Charged is equivalent to Discharging.)
-        * Charge status is Fully Charged only if all batteries are Fully Charged.
+            delay = random.uniform(0, self.config['average_delay'])
+            time.sleep(delay)
 
-        returns: CompositeStatus describing the count and overall charge status of the batteries.
+    def _get_composite_status(self):
+        """Get status information about batteries connected to the device Batwatch is running
+            on, including the charge status, such that:
+            * If any battery is discharging, charge status is Discharging.
+              (Assume any status other than Charging or Fully Charged is equivalent to
+               Discharging.)
+            * Charge status is Fully Charged only if all batteries are Fully Charged.
+
+        returns: CompositeStatus describing the count and overall charge status of the
+            batteries.
         """
 
         device_names = self.upower_bus.EnumerateDevices()
-        charge_status = FULLY_CHARGED
 
         batteries = []
         for device_name in device_names:
             device = self.system_bus.get(UPOWER_BUS_NAME, device_name)
-            if device.PowerSupply is True and device.Type in (2, 3):  # Type 2 = Battery, Type 3 = UPS
+            # We only care about device types 2 and 3, which are batteries and UPS units,
+            #   respectively. The full list can be found here:
+            #   https://upower.freedesktop.org/docs/Device.html#Device:Type
+
+            # A device is considered a power supply if it powers the whole system.
```

Comment:
If we do keep the device.Type check, we should flip around these two comments so that they are in the order of the checks below. Also, the blank line between them is not needed. That's kinda why we do the reverse indent.